app/services/data_ingestion/read.py

```
@@ -49,6 +49,45 @@ def read_pdf_file(file: UploadFile) -> PdfFile:
 
 
+
+def read_paper_pdf_file(file: Union[UploadFile, bytes], filename: str | None = None) -> PdfFile:
+    try:
+        if isinstance(file, UploadFile):
+            raw = file.file.read()
+            name = file.filename
+        elif isinstance(file, bytes):
+            raw = file
+            if not filename:
+                raise ValueError("Filename must be provided for raw bytes")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)